### PR TITLE
[WSP-215] Adds script for searchableType

### PR DESCRIPTION
**Pull request by GeoffreyHayward**

### CODE REVIEW

File: repository-data/application/src/main/resources/hcm-config/configuration/update/WSP-215-add-searchable-type.groovy [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+/**
+ * Adds searchableType true on each document.
+ */
+class SearchableType extends BaseNodeUpdateVisitor {
+
+    boolean doUpdate(Node node) {
+
+        // Query returns the hippo:handle node for the document
+        // (which has the 3 variants)
```
Comment (davidbrrr) [inline repository-data/application/src/main/resources/hcm-config/configuration/update/WSP-215-add-searchable-type.groovy:16]: To be precise, the query will return the variants themselves not the handle.

File: repository-data/application/src/main/resources/hcm-config/configuration/update/WSP-215-add-searchable-type.groovy [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    boolean updateNode(Node n) {
+
+        JcrUtils.ensureIsCheckedOut(n)
+        def path = n.getPath()
+
+        def nodeType = n.getPrimaryNodeType().getName()
+        if (n.hasProperty("common:searchableType")) {
+            log.info("Skipped - Removing publicationsystem:Granularity from node: " + path + " => current node type: " + nodeType)
```
Comment (davidbrrr) [inline repository-data/application/src/main/resources/hcm-config/configuration/update/WSP-215-add-searchable-type.groovy:35]: Log seems to refer to a something else?

File: repository-data/application/src/main/resources/hcm-config/configuration/update/WSP-215-add-searchable-type.groovy [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+            return false
+        } else {
+            log.info("Success - Removing publicationsystem:Granularity from node: " + path + " => current node type: " + nodeType)
```
Comment (davidbrrr) [inline repository-data/application/src/main/resources/hcm-config/configuration/update/WSP-215-add-searchable-type.groovy:38]: Log seems to refer to a something else?

File: repository-data/application/src/main/resources/hcm-config/configuration/update/WSP-215-add-searchable-type.yaml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        or @jcr:primaryType='website:publishedworkchapter' or @jcr:primaryType='website:roadmap'
+        or @jcr:primaryType='website:roadmapitem' or @jcr:primaryType='website:componentlist'
+        or @jcr:primaryType='website:glossarylist' or @jcr:primaryType='website:cyberalert'
+        or @jcr:primaryType='website:blog' or @jcr:primaryType='website:bloghub' or
+        @jcr:primaryType='website:apimaster' or @jcr:primaryType='website:apiendpoint'
+        or @jcr:primaryType='website:apispecification' or @jcr:primaryType='website:person'
+        or @jcr:primaryType='website:businessunit' or @jcr:primaryType='website:orgstructure'
+        or @jcr:primaryType='website:feature' or @jcr:primaryType='website:feedhub')]
+      hipposys:script:
+        resource: /configuration/update/DW-3703-remove-granularity-from-publications.groovy
```
Comment (davidbrrr) [inline repository-data/application/src/main/resources/hcm-config/configuration/update/WSP-215-add-searchable-type.yaml:27]: Shouldn't this be /configuration/update/WSP-215-add-searchable-type.groovy ?

**Review by davidbrrr — CHANGES_REQUESTED** [review]

File: repository-data/application/src/main/resources/hcm-config/configuration/update/WSP-215-add-searchable-type.yaml (hunk 4 above)
Comment (GeoffreyHayward) [reply repository-data/application/src/main/resources/hcm-config/configuration/update/WSP-215-add-searchable-type.yaml:27]: Oh yes. Not sure how that happened. Good spot.

File: repository-data/application/src/main/resources/hcm-config/configuration/update/WSP-215-add-searchable-type.groovy (hunk 1 above)
Comment (GeoffreyHayward) [reply repository-data/application/src/main/resources/hcm-config/configuration/update/WSP-215-add-searchable-type.groovy]: I have updated the comment.

File: repository-data/application/src/main/resources/hcm-config/configuration/update/WSP-215-add-searchable-type.groovy (hunk 2 above)
Comment (GeoffreyHayward) [reply repository-data/application/src/main/resources/hcm-config/configuration/update/WSP-215-add-searchable-type.groovy]: I have updated the message.

File: repository-data/application/src/main/resources/hcm-config/configuration/update/WSP-215-add-searchable-type.groovy (hunk 3 above)
Comment (GeoffreyHayward) [reply repository-data/application/src/main/resources/hcm-config/configuration/update/WSP-215-add-searchable-type.groovy]: I have updated the message.

File: repository-data/application/src/main/resources/hcm-config/configuration/update/WSP-215-add-searchable-type.yaml (hunk 4 above)
Comment (GeoffreyHayward) [reply repository-data/application/src/main/resources/hcm-config/configuration/update/WSP-215-add-searchable-type.yaml]: I have updated the path to the correct script.

**Review by davidbrrr — APPROVED** [review]

**Pull request merged**